storybookjs/storybook
# Core: Fix storySort `order` with whitespace in story paths

Issue: #13311

## What I did

This consistently splits `kind` in a way that allows whitespace around the path separator. This was already implemented in the stories API using `trim()` on each part, but a regex separator is more efficient and less fragile. The `storySort` function didn't do this yet, which is why manually specifying an `order` only worked when including the whitespace in each value (see #13311).

Technically this is a breaking change, because it'll break ordering for people who've specified `order` including whitespace. We could circumvent that by trimming values of `order`. WDYT @shilman ?

## How to test

- Is this testable with Jest or Chromatic screenshots? Updated the unit test
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.
